contracts/token/ERC20/ERC20Channel.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity ^0.4.18;
+
+import "./ERC20.sol";
+import "../../math/SafeMath.sol";
+import "../../ownership/Ownable.sol";
+import "../../ECRecovery.sol";
+
+/**
+   @title ERC20Channel, State channel for ERC20 Tokens
+
+   Contract that provides holders of a ERC20 compatible token the creation of a
+   state channel between two users, once a channel is open the users can exchange
+   signatures offchain to agree on the final value of the transfer.
+   Uses OpenZeppelin ERC20 and SafeMath lib.
+
+   Note: The owner of the contract is the sender, therefore it should be
+    deployed by the sender itself.
+
+   The channel can be closed in two ways: With and agreement or not.
+
+   - Contract closed with an agreement:
+     Can happen at anytime while the channel is still opened. Two signatures
+     are needed for this, one is the signature of the receiver agreeing to
+     receive a certain value, with the receiver signature the sender can now
+     agree on the value and use sign the receiver signature.
+
+      cooperativeClose(
+       FinalBalance,
+       Receiver.sign(FinalBalance),
+       Sender.sign(SHA3( Receiver.sign(FinalBalance) ))
+      )
```

Comment:
I think that Receiver and Sender are swapped here, right?

Reply:
I think it is fine, it is with the dame order n the `uncooperativeClose` function, the first signature is the one form the receiver and the second one from the sender.

Reply:
Hmm I still don't see it. The function is `cooperativeClose(uint256 balance, bytes balanceMsgSig, bytes closingSig)`, where closingSig is the *receiver's* signature of `balanceMsgSig`, which is the *sender's* signature of the hash of the balance. Am I missing sth?